pkg/antctl/transform/addressgroup/transform.go

```diff
@@ -30,11 +35,28 @@ type Response struct {
 }
 
 func listTransform(l interface{}, opts map[string]string) (interface{}, error) {
-	groups := l.(*cpv1beta.AddressGroupList)
-	result := []interface{}{}
-	for i := range groups.Items {
-		item := groups.Items[i]
-		o, _ := objectTransform(&item, opts)
+	groupsList := l.(*cpv1beta.AddressGroupList)
+	if len(groupsList.Items) == 0 {
+		return "", nil
+	}
+	sortField := opts["sort-by"]
+	if sortField == "" {
+		sortField = ".metadata.name"
+	}
+
+	var addressGroupRuntime runtime.Object = &cpv1beta.AddressGroupList{
+		Items: groupsList.Items,
+	}
```

Comment:
I wonder why creating another AddressGroupList here? Isn't groupsList already a AddressGroupList?

Reply:
the sortObjects function accepts only []runtime.Object as argument and groupList is not of that type, its of the type cpv1beta.NetworkPolicyList, so we fist need to convert this list into runtime.Object nad then into slice of runtime.Object

Reply:
I understand that part.
I think you create `addressGroupRuntime` for `meta.ExtractList`, right? `groupsList` itself can be used in `meta.ExtractList`, then why we need `addressGroupRuntime`?

Reply:
okay got it